lib/commands/utility.go

```diff
@@ -1,19 +1,189 @@
+				color.Red("Pushing these will overwrite the API favorites with the local settings.\n\n")
+			}
+
+			selection, err := helper.PromptSelect("\nDo you want to push your local favorites to the Kion API?", []string{"no", "yes"})
+			if err != nil {
+				fmt.Printf("Error prompting for confirmation: %v\n", err)
+				return nil
+			}
+			if selection == "no" {
+				fmt.Println("Aborting push of favorites.")
+				return nil
+			}
+
+			if len(result.Conflicts) > 0 {
+				confirm, err := helper.PromptSelect("You have some name conflicts with API favorites. Are you sure you want to continue pushing your local favorites?", []string{"no", "yes"})
+				if err != nil {
+					return err
+				}
+				if confirm == "no" {
+					fmt.Println("Aborting push of favorites due to conflicts.")
+					return nil
+				}
+			}
+
+			fmt.Println("Pushing local favorites to the Kion API...")
+
+			for _, f := range result.LocalOnly {
+				if len(f.Name) > 50 {
+					color.Yellow("Trimming favorite %s because its name exceeds 50 characters.\n", f.Name)
+					f.Name = f.Name[:50]
+				}
+
+				// set access type to match Kion API requirements
+				if f.AccessType == "web" {
+					f.AccessType = "console_access"
+				}
+				if f.AccessType == "cli" {
+					f.AccessType = "short_term_key_access"
+				}
+
+				newFav, status, err := kion.CreateFavorite(c.config.Kion.Url, c.config.Kion.ApiKey, f)
+				if err != nil {
+					color.Red("Error creating favorite %s: %v\n", f.Name, err)
+					continue
+				}
+				if status == 201 || status == 200 {
+					color.Green("Successfully created favorite: %s\n", newFav)
+				} else {
+					color.Red("Failed to create favorite %s, status code: %d\n", f.Name, status)
+				}
+			}
+
+			for _, f := range result.Conflicts {
+
+				// set access type to match Kion API requirements
+				if f.AccessType == "web" {
+					f.AccessType = "console_access"
+				}
+				if f.AccessType == "cli" {
+					f.AccessType = "short_term_key_access"
+				}
+
+				_, err := kion.DeleteFavorite(c.config.Kion.Url, c.config.Kion.ApiKey, f.Name)
+				if err != nil {
+					color.Red("Error deleting favorite %s: %v\n", f.Name, err)
+					continue
+				}
+				color.Green("Successfully deleted conflicting favorite: %s\n", f.Name)
+
+				_, _, err = kion.CreateFavorite(c.config.Kion.Url, c.config.Kion.ApiKey, f)
+				if err != nil {
+					color.Red("Error creating favorite %s: %v\n", f.Name, err)
+					continue
+				}
+				color.Green("Successfully created favorite: %s\n", f.Name)
+			}
+
+			// send to the DeleteLocalFavorites function to remove local favorites after successful push
+			return c.DeleteLocalFavorites(configPath)(cCtx)
+		} else {
+			color.Yellow("Favorites API is not enabled. This requires Kion version 3.13.0 or higher.")
+		}
+		return nil
+	}
+}
+
+func (c *Cmd) DeleteLocalFavorites(configPath string) func(cCtx *cli.Context) error {
```

Comment:
If we redo the func signature above we should do the same here, using configPath as a cCtx metadata variable.